piccolo_api/crud/endpoints.py

```diff
@@ -1060,6 +1067,12 @@ async def patch_single(
             await cls.update(values).where(
                 cls._meta.primary_key == row_id
             ).run()
+            try:
+                await AuditLog.post_patch_action(
+                    cls, row_id=row_id, user_id=request.user.user_id
+                )
```

Comment:
I think it should all be optional. So PiccoloCRUD should have a new argument which is `audit_log: t.Optional[t.Type[AuditLog]] = None`. And then this line would be:

```python
if self.audit_table:
    await AuditLog.post_patch_action(
        cls, row_id=row_id, user_id=request.user.user_id
     )
```

Otherwise when we release this it will be a breaking change - anyone without the `AuditLog` table installed will have their code break. Also, it allows people to disable this functionality if they don't want to track changes.

Reply:
You're right. I will change that.